custom_components/tahoma/coordinator.py

```diff
@@ -62,17 +62,45 @@ async def _async_update_data(self) -> Dict[str, Device]:
             raise UpdateFailed(exception)
 
         for event in events:
-            if event.name == "DeviceStateChangedEvent":
+            _LOGGER.debug(
+                f"{event.name}/{event.exec_id} (device:{event.deviceurl},state:{event.old_state}->{event.new_state})"
+            )
+
+            if event.name == "DeviceAvailableEvent":
+                self.devices[event.deviceurl].available = True
+
+            elif event.name == "DeviceUnavailableEvent":
+                self.devices[event.deviceurl].available = False
+
+            elif event.name == "DeviceStateChangedEvent":
                 for state in event.device_states:
                     device = self.devices[event.deviceurl]
                     if state.name not in device.states:
                         device.states[state.name] = state
                     device.states[state.name].value = self._get_state(state)
 
+            if (
+                event.name == "ExecutionRegisteredEvent"
+                and event.exec_id in self.executions
```

Comment:
Finally, do we care if the command has been triggered by HA? Imagine I launch it from the tahoma website, it will be nice to have the update asap in HA too.

Reply:
True! I made the changes for this, but we would need to test it.